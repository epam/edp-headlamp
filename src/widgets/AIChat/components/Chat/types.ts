import { CodemieSecretData, ConversationItem } from '../../types';

export interface ChatProps {
  codemieSecretData: CodemieSecretData;
  conversation: ConversationItem;
  updateConversation: (conversation: ConversationItem) => void;
  requestError: Error;
}
