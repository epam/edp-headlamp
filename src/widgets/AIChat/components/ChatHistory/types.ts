import { ConversationItem, ConversationState } from '../../types';

export interface ChatHistoryProps {
  conversations: ConversationItem[];
  activeConversation: ConversationItem | null;
  setActiveConversation: React.Dispatch<React.SetStateAction<ConversationItem | null>>;
  setConversations: React.Dispatch<React.SetStateAction<ConversationState>>;
  createNewConversation: () => void;
}
