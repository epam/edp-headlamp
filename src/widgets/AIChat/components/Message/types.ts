import React from 'react';
import { ChatEntity } from '../../types';

export interface MessageProps {
  entityRole: ChatEntity;
  createdAt: string;
  content: React.ReactNode | string;
}
