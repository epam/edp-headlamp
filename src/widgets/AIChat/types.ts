import { ValueOf } from '../../types/global';
import { CHAT_ENTITY } from './constants';

export type ChatEntity = ValueOf<typeof CHAT_ENTITY>;

export interface ChatItemRequest {
  message: string;
}

export interface ProcessedChunk {
  id: string;
  text: string;
  isCode: boolean;
}

export interface ChatItemResponse {
  message: string | undefined; // is available on streaming finish
  processedChunks: ProcessedChunk[];
}

export interface HistoryStateItem {
  id: string;
  createdAt: string;
  request: ChatItemRequest;
  response?: ChatItemResponse;
}

export interface HistoryPayloadItem {
  createdAt: string;
  message: string;
  role: ChatEntity;
}

export interface ConversationItem {
  conversationId: string;
  conversationHistory: HistoryStateItem[];
  assistantID: string;
  assistantName: string;
  assistantContext: string;
  prompt: string;
  llmModel: string;
  assistantMode: boolean;
  file: File | null;
  stream: boolean;
  _debug: boolean;
  topK: number;
  codeMode?: boolean;
  kbMode?: boolean;
  kbType?: string;
  kbName?: string;
  appName?: string;
  repoName?: string;
  indexType?: string;
}

export interface ConversationPayload {
  conversationId: string;
  text: string;
  file: File | null;
  prompt: string;
  history: HistoryPayloadItem[];
  llmModel: string;
  stream: boolean;
  _debug: boolean;
  topK: number;
  code_fields?: {
    appName: string;
    repoName: string;
    indexType: string;
  };
}

export interface ConversationResponseChunk {
  time_elapsed: number;
  input_tokens_used: number;
  tokens_used: number;
  generated_chunk: string;
  generated: string;
  thought: string;
  last: boolean;
  debug: null;
}

export interface ConversationState {
  [key: string]: ConversationItem;
}

export interface CodemieSecretData {
  apiUrl: string;
  assistantId: string;
  token: string;
}
