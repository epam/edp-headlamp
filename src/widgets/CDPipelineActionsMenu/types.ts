import { ACTION_MENU_TYPES } from '../../constants/actionMenuTypes';
import { CDPipelineKubeObjectInterface } from '../../k8s/groups/EDP/CDPipeline/types';
import { ValueOf } from '../../types/global';
import { PermissionSet } from '../../types/permissions';

export interface CDPipelineActionsMenuProps {
  data: {
    CDPipelineData: CDPipelineKubeObjectInterface;
  };
  backRoute?: string;
  variant?: ValueOf<typeof ACTION_MENU_TYPES>;
  anchorEl?: HTMLElement;
  handleCloseResourceActionListMenu?: () => void;
  permissions: PermissionSet;
}
