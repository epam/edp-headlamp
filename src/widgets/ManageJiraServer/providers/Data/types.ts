import { SecretKubeObjectInterface } from '../../../../k8s/groups/default/Secret/types';
import { JiraServerKubeObjectInterface } from '../../../../k8s/groups/EDP/JiraServer/types';

export interface DataContextProviderValue {
  secret: SecretKubeObjectInterface;
  jiraServer: JiraServerKubeObjectInterface;
  ownerReference: string | undefined;
  handleClosePanel?: () => void;
}

export interface DataContextProviderProps {
  secret: SecretKubeObjectInterface;
  jiraServer: JiraServerKubeObjectInterface;
  ownerReference: string | undefined;
  handleClosePanel?: () => void;
}
