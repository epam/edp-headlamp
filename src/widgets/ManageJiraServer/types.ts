import { SecretKubeObjectInterface } from '../../k8s/groups/default/Secret/types';
import { JiraServerKubeObjectInterface } from '../../k8s/groups/EDP/JiraServer/types';
import { FormValues } from '../../types/forms';
import { ValueOf } from '../../types/global';
import { FORM_NAMES } from './constants';
import { INTEGRATION_SECRET_FORM_NAMES, JIRA_SERVER_FORM_NAMES } from './names';

export type FormNames = ValueOf<typeof FORM_NAMES>;

export interface ManageJiraServerProps {
  secret: SecretKubeObjectInterface;
  jiraServer: JiraServerKubeObjectInterface;
  ownerReference: string | undefined;
  handleClosePanel?: () => void;
}

export type JiraServerFormValues = FormValues<typeof JIRA_SERVER_FORM_NAMES>;

export type IntegrationSecretFormValues = FormValues<typeof INTEGRATION_SECRET_FORM_NAMES>;
