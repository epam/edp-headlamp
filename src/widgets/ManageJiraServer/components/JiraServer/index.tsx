import { Stack } from '@mui/material';
import React from 'react';
import { URL } from './fields';

export const JiraServerForm = () => {
  return (
    <Stack spacing={2}>
      <URL />
    </Stack>
  );
};
