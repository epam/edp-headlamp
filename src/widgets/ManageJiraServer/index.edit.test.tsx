/**
 * @jest-environment jsdom
 */

import { render, screen } from '@testing-library/react';
import React from 'react';
import { TestWrapper } from '../../../mocks/wrappers/default';
import { JiraCISecretMock } from '../../k8s/groups/default/Secret/mocks/jira-ci-secret.mock';
import { SecretKubeObjectInterface } from '../../k8s/groups/default/Secret/types';
import { JiraServerMock } from '../../k8s/groups/EDP/JiraServer/mocks/jira-server.mock';
import { JiraServerKubeObjectInterface } from '../../k8s/groups/EDP/JiraServer/types';
import { ManageJiraServer } from './index';

test('renders ManageJiraServer Edit component', () => {
  render(
    <TestWrapper>
      <ManageJiraServer
        secret={JiraCISecretMock as unknown as SecretKubeObjectInterface}
        jiraServer={JiraServerMock as JiraServerKubeObjectInterface}
        ownerReference={null}
        handleClosePanel={jest.fn()}
      />
    </TestWrapper>
  );

  const dialog = screen.getByTestId('form');
  expect(dialog).toMatchSnapshot();
});
