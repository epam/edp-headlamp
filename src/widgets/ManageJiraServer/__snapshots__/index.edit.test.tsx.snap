// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ManageJiraServer Edit component 1`] = `
<div
  data-testid="form"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        class="MuiStack-root css-1sazv7p-MuiStack-root"
      >
        <div
          class="MuiStack-root css-jfdv4h-MuiStack-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-98a9yt-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for="mui-2"
                id="mui-2-label"
              >
                URL
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1wqgp0s-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-2"
                  name="url"
                  placeholder="Enter Jira URL"
                  type="text"
                  value="https://test-jira.com"
                />
                <span />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-14lo706"
                  >
                    <span>
                      URL
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
        >
          <div
            class="MuiStack-root css-jfdv4h-MuiStack-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-98a9yt-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for="mui-4"
                  id="mui-4-label"
                >
                  User
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1wqgp0s-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                    id="mui-4"
                    name="username"
                    placeholder="Enter user name"
                    type="text"
                    value="test-username"
                  />
                  <span />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-14lo706"
                    >
                      <span>
                        User
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
        >
          <div
            class="MuiStack-root css-jfdv4h-MuiStack-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-98a9yt-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for="mui-6"
                  id="mui-6-label"
                >
                  Password
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1wqgp0s-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="off"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                    id="mui-6"
                    name="password"
                    placeholder="Enter password"
                    type="password"
                    value="test-password"
                  />
                  <div
                    class="MuiStack-root css-niqf4j-MuiStack-root"
                  >
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                    >
                      <span />
                    </div>
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                    >
                      <button
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <span />
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-14lo706"
                    >
                      <span>
                        Password
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        class="MuiStack-root css-1a3p8mp-MuiStack-root"
      >
        <div
          class="MuiBox-root css-1po99kh"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-vw8gqz-MuiButtonBase-root-MuiButton-root"
            style="pointer-events: auto;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
            >
              <span />
            </span>
            Reset integration
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiStack-root css-pl8gqh-MuiStack-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-1llbb2a-MuiButtonBase-root-MuiButton-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            undo changes
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-vw8gqz-MuiButtonBase-root-MuiButton-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            save
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
